File: from-1501-to-1600/1524-number-of-sub-arrays-with-odd-sum/solution.go
func numOfSubarrays(arr []int) int {
    MOD := 1000000007
    oddity := make([]int, len(arr))
    curr := 0
    for i, num := range arr {
        curr = (curr + num) % 2
        oddity[i] = curr
    }

    odd, even := 0, 0
    result := 0

    for _, v := range oddity {
        if v == 0 {
            even += 1
            result += odd
        } else {
            odd += 1
            result += even + 1
        }
        result %= MOD
    }
    return result
}

File: from-1501-to-1600/1524-number-of-sub-arrays-with-odd-sum/solution.py
class Solution:
    def numOfSubarrays(self, arr: List[int]) -> int:
        MOD = 1000000007
        oddity = [0] * len(arr)
        curr = 0
        for i, num in enumerate(arr):
            curr = (curr + num) % 2
            oddity[i] = curr

        odd, even = 0, 0
        result = 0

        for v in oddity:
            if v == 0:
                even += 1
                result += odd
            else:
                odd += 1
                result += even + 1
            
            result %= MOD

        return result